File: TME.Store/TME.Store.Domain/Models/SearchProductsResult.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TME.Store.Domain.Models
{
    public class SearchProductsResult : ProductsResult
    {
        public int PageCount { get; private set; }

        public SearchProductsResult(string priceType, string currency, IList<Product> products, int pageCount) 
            : base(priceType, currency, products)
        {
            PageCount = pageCount;
        }
    }
}

File: tme_store/lib/models/search_products_result.dart
import 'package:tme_store/models/models.dart';

class SearchProductsResult {
  final String priceType;
  final String currency;
  final List<Product> products;
  final int totalCount;
  final Map<String, int> totalProductsPerCategory;

  SearchProductsResult({
    this.priceType,
    this.currency,
    this.products,
    this.totalCount,
    this.totalProductsPerCategory
  });

  static SearchProductsResult empty() {
    return SearchProductsResult(
      currency: null, 
      priceType: null,
      products: [],
      totalCount: 0);
  }
}